rust/pact_ffi/src/mock_server/handles.rs

```diff
@@ -1650,185 +1650,224 @@ pub extern fn pactffi_response_status_v2(interaction: InteractionHandle, status:
-/// part parameter will be ignored. With synchronous messages, the request contents will be overwritten,
-/// while a new response will be appended to the message.
+/// The content type is obtained from a number of sources, in the following
+/// order:
+///
+/// - The `content_type` parameter, if provided; or,
+/// - The `content_type_hint` parameter, if provided (e.g. from the
+///   `Content-Type` header); or,
+/// - From the body itself; or,
+/// - Defaults to `text/plain` as a last resort.
+fn process_body(
+    body: &str,
+    content_type: &Option<ContentType>,
+    content_type_hint: &Option<ContentType>,
+    matching_rules: &mut MatchingRules,
+    generators: &mut Generators,
+) -> OptionalBody {
+    trace!(
+        ">>> process_body({:?}, {:?}, {:?}, {:?}, {:?})",
+        body,
+        content_type,
+        content_type_hint,
+        matching_rules,
+        generators
+    );
+    let detected_type = detect_content_type_from_string(body);
+    let content_type = content_type
+        .clone()
+        .or_else(|| content_type_hint.clone())
+        .or_else(|| detected_type.clone())
+        .or_else(|| Some(TEXT.clone()));
+    trace!(?detected_type);
+    trace!(?content_type);
+
+    match content_type {
+        Some(ct) if ct.is_json() => {
+            trace!("Processing JSON body");
+            // As the JSON body may contain embedded generators and matching
+            // rules, we must use `process_json` to ensure they are processed
+            // correctly and added to the `generators` and `matching_rules`.
+            let category = matching_rules.add_category("body");
+            OptionalBody::Present(
+                Bytes::from(process_json(body.to_string(), category, generators)),
+                Some(ct),
+                None,
+            )
+        }
+        Some(ct) if ct.is_xml() => {
+            // The XML payload may contain one of two cases:
+            // 1. A raw XML payload
+            // 2. A JSON payload describing the XML payload, including any
+            //    embedded generators and matching rules.
+            match detected_type {
+                Some(detected_ct) if detected_ct.is_json() => {
+                    trace!("Processing JSON description for XML body");
+                    let category = matching_rules.add_category("body");
+                    OptionalBody::Present(
+                        Bytes::from(
+                            process_xml(body.to_string(), category, generators).unwrap_or_default(),
+                        ),
+                        Some(ct), // Note to use the provided content type, not the detected one
+                        None,
+                    )
+                }
+                _ => {
+                    trace!("Raw XML body left as is");
+                    OptionalBody::from(body)
+                }
+            }
+        }
+        None | Some(_) => {
```

Comment:
This can just be abbreviated as `_ => {`

Reply:
Accidental left-over of some refactoring. Fixed :)